fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexer.java

```diff
@@ -314,6 +331,38 @@ public void close() {
         return ret;
     }
 
+    /**
+     * Run after a single transactional call within runAsync.
+     */
+    private <R> Pair<R, Throwable> runAsyncPostTransaction(R result, Throwable exception) {
+        if (exception == null) {
+            successCount++;
+            if (successCount >= increaseLimitAfter && limit < maxLimit) {
+                increaseLimit();
```

Comment:
Should this set `successCount` back to zero after increasing the limit, or is the intention to keep increasing the limit until we start running into problems again (or hit the limit limit)?

Reply:
Yes, the intention is to have it go right back up until it hit the limit or started hitting issues again. There's definitely going to be an art to balancing the different parameters here, but without a large amount of engineering, I think that's the best we can do.

Reply:
Okay, sounds good. Seems like something we could tweak later.